lib/active_fedora/clean_connection.rb

```diff
@@ -0,0 +1,35 @@
+module ActiveFedora
+  class CleanConnection < SimpleDelegator
+    def get(*args)
+      result = __getobj__.get(*args) do |req|
+        prefer_headers = Ldp::PreferHeaders.new(req.headers["Prefer"])
+        prefer_headers.omit = prefer_headers.omit | omit_uris
+        req.headers["Prefer"] = prefer_headers.to_s
+      end
+      CleanResult.new(result)
+    end
+
+    private
+
+    def omit_uris
+      [
+        RDF::Fcrepo4.ServerManaged,
+        RDF::Ldp.PreferContainment,
+        RDF::Ldp.PreferEmptyContainer,
+        RDF::Ldp.PreferMembership
+      ]
+    end
+
+    class CleanResult < SimpleDelegator
+      def graph
+        __getobj__.graph.delete(has_model_query)
```

Comment:
@jeremyf Look better? Thought it would be good for it not to continuously rerun the delete as well.

One thing that's interesting is graph#delete returning a graph without the statement is similar to how a functional language would do it, no? It -does- mutate the underlying object, but I wonder if the return value was intentional. https://github.com/ruby-rdf/rdf/blob/develop/lib/rdf/mixin/mutable.rb#L137 looks like it is, but immutability certainly doesn't seem to be the goal given the name of the mixin.